src/main/scala/characters/Person.scala

```diff
@@ -0,0 +1,12 @@
+package characters
+
+case class Person(
+                   name: String,
+                   gender: String,
+                   hairColor: String,
+                   wearsGlasses: Boolean,
+                   wearsHat: Boolean,
+                   hasBeard: Boolean,
+                   eyeColor: String,
+                 )
```

Comment:
You could of also used traits where there are over all types with sub types. 
E.g. `trait Gender
  case object Female extends Gender
  case object Male extends Gender
  case object NonBinary extends Gender`
  A `Person` can then have the trait Gender as the data type for gender parameter (`gender: Gender`) and this will now accept them having any of the case objects that extends this trait.
  An example of this could be a method that accepts the trait as a parameter and then accepts you passing in Female, Male or NonBinary which all extend this trait being given to the method. 
  E.g.
  `def stringOfCaseObject(value: Gender): String = value.toString

  println(stringOfCaseObject(Female))`

Reply:
Created seperate case classes for wearsHat, wearsGlasses, hasBeard and created Traits for Gender, EyeColour.